services/workflows-service/src/workflow/cron/ongoing-monitoring.cron.ts

```diff
@@ -77,27 +82,40 @@ export class OngoingMonitoringCron {
               continue;
             }
 
-            const lastReportFinishedDate = lastReceivedReport.createdAt;
-            const dateToRunReport = new Date(
-              new Date().setTime(
-                lastReportFinishedDate.getTime() + intervalInDays * 24 * 60 * 60 * 1000,
-              ),
-            );
+            const now = new Date().getTime();
+            const options = featureConfig.options;
+
+            const lastReceivedReportTime = lastReceivedReport.createdAt.getTime();
+
+            if (
+              (options.scheduleType === 'specific' &&
+                new Date().getDate() !== options.specificDates.dayInMonth &&
+                lastReceivedReportTime + 32 * ONE_DAY < now) ||
+              (options.scheduleType === 'interval' &&
+                now - lastReceivedReportTime < options.intervalInDays * ONE_DAY)
+            ) {
+              this.logger.debug(`Skipping ongoing report for business: ${business.id}`);
 
-            if (dateToRunReport <= new Date()) {
-              await this.invokeOngoingAuditReport({
-                business: business as Business & {
-                  metadata?: { featureConfig?: Record<string, TCustomerFeatures> };
-                },
-                workflowDefinitionConfig: processConfig,
-                workflowDefinitionId: workflowDefinition.id,
-                currentProjectId: business.projectId,
-                projectIds: projectIds,
-                lastReportId: lastReceivedReport.reportId,
-                checkTypes: processConfig?.checkTypes,
-                reportType: this.processFeatureName,
-              });
+              continue;
+            }
+
+            if (ongoingReportsCounter >= env.ONGOING_REPORTS_LIMIT) {
+              this.logger.debug(`Ongoing reports limit reached for current run`);
```

Comment:
Log without a context is log that we can't do anything with is. At least add the customer ID?

Reply:
No need. It indicates that we've maximized the number of reports for the cron run, next cron run will continue